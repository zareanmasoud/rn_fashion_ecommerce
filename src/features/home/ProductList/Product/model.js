import {PRODUCT_SCREEN} from 'screens/routes/constants';

const model = {
  product(navigation, data) {
    return {
      press: () => navigation.navigate(PRODUCT_SCREEN, {id: data.id}),
      photo: data.photo !== null ? data.photo : '',
    };
  },
};

export default model;
