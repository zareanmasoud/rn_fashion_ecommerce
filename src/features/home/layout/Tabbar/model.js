import R from 'res/R';

const model = {
  buttons: {
    home: {
      id: 0,
      handlePress: () => {},
      src: R.images.backButton,
      text: R.strings.home.layout.mainTabbar.home,
    },
    wishlist: {
      id: 1,
      handlePress: () => {},
      src: R.images.cartButton,
      text: R.strings.home.layout.mainTabbar.wishlist,
    },
    help: {
      id: 2,
      handlePress: () => {},
      src: R.images.logoButton,
      text: R.strings.home.layout.mainTabbar.help,
    },
    profile: {
      id: 3,
      handlePress: () => {},
      src: R.images.logoButton,
      text: R.strings.home.layout.mainTabbar.profile,
    },
  },
};

export default model;
