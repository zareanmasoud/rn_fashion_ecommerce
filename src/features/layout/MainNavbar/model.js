import R from 'res/R';

const model = {
  backButton(navigation) {
    return {
      press: () => navigation.pop(),
      src: R.images.backButton,
    };
  },
  buttonTwo(cart, cartTextTestID) {
    return {
      press: () => {},
      src: R.images.cartButton,
      badge: {
        cart,
        cartTextTestID,
      },
    };
  },
  logoButton: {
    press: () => {},
    src: R.images.logoButton,
  },
};

export default model;
