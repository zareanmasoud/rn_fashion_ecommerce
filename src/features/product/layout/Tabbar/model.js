import R from 'res/R';

const model = {
  buttons: {
    addWishlist: {
      id: 1,
      press: () => {},
      src: R.images.backButton,
      text: R.strings.home.layout.mainTabbar.home,
    },
    fitting: {
      id: 2,
      press: () => {},
      src: R.images.cartButton,
      text: R.strings.home.layout.mainTabbar.wishlist,
    },
    addToCart(press) {
      return {
        id: 3,
        press: () => press(),
        src: R.images.logoButton,
        text: R.strings.home.layout.mainTabbar.help,
      };
    },
  },
};

export default model;
