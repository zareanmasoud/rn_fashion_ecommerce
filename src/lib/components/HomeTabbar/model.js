// @flow

export type Buttons = {
  press: Function,
  src: string,
  text: string,
};
