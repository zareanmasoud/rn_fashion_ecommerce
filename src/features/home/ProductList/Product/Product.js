import React, {useContext} from 'react';
import Item from 'lib/components/MyFlatList/Item';
import type {Data} from 'lib/components/MyFlatList/Item/types';
import {NavigationContext} from 'react-navigation';
import model from './model';

type Props = {
  data: Data,
  index: number,
};

const Product = ({data, index}: Props) => {
  const navigation = useContext(NavigationContext);
  return (
    <Item key={index} index={index} {...model.product(navigation, data)} />
  );
};

export default Product;
