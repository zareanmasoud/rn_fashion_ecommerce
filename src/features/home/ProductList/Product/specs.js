import {PRODUCT_SCREEN} from 'screens/routes/constants';

const specs = {
  spec(navigation, data) {
    return {
      press: () => navigation.navigate(PRODUCT_SCREEN, {id: data.id}),
    };
  },
};

export default specs;
