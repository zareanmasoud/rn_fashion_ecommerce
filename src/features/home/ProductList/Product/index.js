import Product from './Product';

export {Product as default};
