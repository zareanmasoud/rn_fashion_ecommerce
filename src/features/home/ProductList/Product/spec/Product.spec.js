import React from 'react';
import {shallow} from 'enzyme';
import Product from '../Product';
import model from './model';

describe('Product Feature', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const wrapper = shallow(<Product data={model.product} index={0} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
