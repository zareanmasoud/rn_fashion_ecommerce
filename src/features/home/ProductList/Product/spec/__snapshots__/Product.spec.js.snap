// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Feature Rendering should match to snapshot 1`] = `
<Item
  handlePress={[Function]}
  index={0}
  key="0"
  photo={
    Object {
      "process": [Function],
    }
  }
/>
`;
